rpc/handlers_test.go

```diff
@@ -162,6 +163,18 @@ func TestBlockTransactionCount(t *testing.T) {
 		require.Nil(t, rpcErr)
 		assert.Equal(t, expectedCount, count)
 	})
+
+	t.Run("blockID - pending", func(t *testing.T) {
+		latestBlock.Hash = nil
+		latestBlock.GlobalStateRoot = nil
+		mockReader.EXPECT().Pending().Return(blockchain.Pending{
+			Block: latestBlock,
```

Comment:
I don't see you setting `latestBlock.Number = nil`, am I missing something?

Reply:
`latestBlock.Number` is a `uint64`, not a pointer. So any value you can set it to doesnt really signal if it is a pending or canonical block.